### ISSUE TITLE
Feature Request: `AWS::SSM::Parameter` - Improve the error message for `/aws*` parameters

### ISSUE THREAD

**Issue by r-heimann** [aws-cloudformation/cfn-lint#4089]
### Is this feature request related to a new rule or cfn-lint capabilities?

New capability

### Describe the feature you'd like to request

cfn-lint correctly checks via RegEx if a parameter starts with `/aws`, but it is unclear in the error message why this parameter cannot be used:

<img width="696" alt="Image" src="https://github.com/user-attachments/assets/86fd5f0c-a883-4dee-a4ee-697d4d47ae12" />

It would be great if the error message could be improved to make it clear that a user cannot create `/aws*` parameters. This is the current error message in the AWS Console for Systems Manager Parameters:

`No access to reserved parameter name: aws/test.`

### Describe the solution you'd like

A better error message for users. Example template:

```yaml
AWSTemplateFormatVersion: 2010-09-09
Description: Test

Resources:
  SNSTopic:
    Type: AWS::SNS::Topic

  SNSTopicParameter:
    Type: AWS::SSM::Parameter
    Properties:
      Name: "/aws/test"
      Type: String
      Value: !GetAtt SNSTopic.TopicArn
```

### Additional context

The error message in CloudFormation/Parameters API? is useless, so having a good error description in cfn-lint would be great:

```
Resource handler returned message:
"Error occurred during operation 'PutParameter'."
(RequestToken: 1...c, HandlerErrorCode: GeneralServiceException)
```

### Is this something that you'd be interested in working on?

- [ ] 👋 I may be able to implement this feature request

### Would this feature include a breaking change?

- [ ] ⚠️ This feature might incur a breaking change
